Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs

```diff
@@ -647,6 +654,29 @@ private async Task<BoolResult> RestoreCheckpointStateAsync(OperationContext cont
                         Task.Run(() => ReconcileAsync(context), context.Token).FireAndForget(context);
                     }
                 }
+
+                // Make sure to only have one thread doing proactive replication, in case last proactive replication is still happening.
+                if (_configuration.EnableProactiveReplication && Interlocked.CompareExchange(ref _replicating, 1, 0) == 0)

```

Comment:
I think we should use another patter here, when we have a task field and a lock and we just returning an existing task if its still running.

I think this place is not a perf critical enough to deserve lock free impl. I would suggest using more readable solution.